### ISSUE TITLE
Preprocessing fails when calling _build_vocabs() - inconsistent method signature

### ISSUE THREAD

**Issue by mmartin9684-sil** [sillsdev/silnlp#194]
Running the preprocessing command with a HF model (e.g., NLLB) fails on a call to the _build_vocabs() method.  The signature for the abstract method (1 parameter == 'self') differs from the signature for the HF implementation of the method (2 parameters == 'self' and 'stats').

Stack trace below.

Also, it should be possible to calculate src/trg alignment statistics with both implementations of the method (HF and OpenNMT), but only the HF implementation accepts the 'stats' flag to enable/disable this feature.

`C:\Users\mm9q\AppData\Local\pypoetry\Cache\virtualenvs\silnlp-tgFQQ4V2-py3.8\Scripts\python.exe -m silnlp.nmt.preprocess FT-Philippines\Alignments --stats 
2023-08-30 14:07:29.345895: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2023-08-30 14:07:29.346142: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2023-08-30 14:07:35,239 - silnlp.common.environment - INFO - Using workspace: S:/ as per environment variable SIL_NLP_DATA_PATH.
2023-08-30 14:07:38,210 - silnlp.common.utils - INFO - Git commit: ff4de38ad0
Traceback (most recent call last):
  File "C:\Users\mm9q\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\mm9q\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\mm9q\PycharmProjects\silnlp\silnlp\nmt\preprocess.py", line 29, in <module>
    main()
  File "C:\Users\mm9q\PycharmProjects\silnlp\silnlp\nmt\preprocess.py", line 24, in main
    config.preprocess(args.stats)
  File "C:\Users\mm9q\PycharmProjects\silnlp\silnlp\nmt\config.py", line 468, in preprocess
    self._build_vocabs(stats)
TypeError: _build_vocabs() takes 1 positional argument but 2 were given

Process finished with exit code 1`

**Comment by mshannon-sil** [comment]
I just submitted a pull request to make the function signatures for `_build_vocabs()` consistent. I ran the experiment listed above with that change and it looks to have fixed the issue.

As for calculating alignment, `_build_vocabs()` does not calculate alignment when stats is set to `True`; it only calculates stats directly related to tokenization. It is `_build_corpora()` that calculates alignment when stats is set to `True`. `build_corpora()` is defined in config.py, and both OpenNMTConfig and HuggingFaceConfig use this definition.

Also, when I took a look at the config.yml for the `FT-Philippines\Alignments`, there was no model specified, which is why it's defaulting to the OpenNMTConfig class.
